packages/core/src/lib/markdown-it/highlight/HighlightRule.ts

```diff
@@ -48,10 +52,16 @@ class HighlightRule {
       return null;
     }
 
-    return new HighlightRule(components);
+    const componentsWithoutNull: HighlightRuleComponent[] = [];
+    components.forEach((c: HighlightRuleComponent | null) => {
```

Comment:
No sadly.. the type is still `(HighlightRuleComponent | null) []` after filtering
By right, there shouldn't be any null anyway because of the previous check `some(c => !c)` but this is not recognised by the linter

Reply:
Ah I see, thanks for the clarification 😭 In that case could we do a typecast directly in the return (`component as HighlightRuleComponent[]`) since we're sure that there shouldn't be nulls?